http-server/src/test/java/xtdb/jackson/JsonLdValueOrPersistentHashMapDeserializerTest.java

```diff
@@ -0,0 +1,74 @@
+package xtdb.jackson;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import clojure.java.api.Clojure;
+import clojure.lang.*;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.module.SimpleDeserializers;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+class JsonLdValueOrPersistentHashMapDeserializerTest {
+
+    private final ObjectMapper objectMapper;
+    private IFn keywordFn = Clojure.var("clojure.core", "keyword");
+    private IFn setFn = Clojure.var("clojure.core", "set");
+
+    public JsonLdValueOrPersistentHashMapDeserializerTest() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true);
+        SimpleModule module = new SimpleModule("JsonLd-test");
+
+        HashMap<String, IFn> decoders = new HashMap<String, IFn>();
```

Comment:
Given most (all?) of the encoders/decoders are Java methods, is it worth us having these written in Java, so that (again) you're testing real code here rather than having to replicate it in the tests?

Reply:
I'd prefer the IFn here as I would like to be able to easily extend this without writing an Java.